pulsar-client-api/src/main/java/org/apache/pulsar/client/api/PulsarClientException.java

```diff
@@ -957,77 +1003,110 @@ public static PulsarClientException unwrap(Throwable t) {
         } else if (cause instanceof InvalidTopicNameException) {
-            return new InvalidTopicNameException(msg);
+            newException = new InvalidTopicNameException(msg);
         } else if (cause instanceof NotSupportedException) {
-            return new NotSupportedException(msg);
+            newException = new NotSupportedException(msg);
         } else if (cause instanceof NotAllowedException) {
-            return new NotAllowedException(msg);
+            newException = new NotAllowedException(msg);
         } else if (cause instanceof ProducerQueueIsFullError) {
-            return new ProducerQueueIsFullError(msg);
+            newException = new ProducerQueueIsFullError(msg);
         } else if (cause instanceof ProducerBlockedQuotaExceededError) {
-            return new ProducerBlockedQuotaExceededError(msg);
+            newException = new ProducerBlockedQuotaExceededError(msg);
         } else if (cause instanceof ProducerBlockedQuotaExceededException) {
-            return new ProducerBlockedQuotaExceededException(msg);
+            newException = new ProducerBlockedQuotaExceededException(msg);
         } else if (cause instanceof ChecksumException) {
-            return new ChecksumException(msg);
+            newException = new ChecksumException(msg);
         } else if (cause instanceof CryptoException) {
-            return new CryptoException(msg);
+            newException = new CryptoException(msg);
         } else if (cause instanceof ConsumerAssignException) {
-            return new ConsumerAssignException(msg);
+            newException = new ConsumerAssignException(msg);
         } else if (cause instanceof MessageAcknowledgeException) {
-            return new MessageAcknowledgeException(msg);
+            newException = new MessageAcknowledgeException(msg);
         } else if (cause instanceof TransactionConflictException) {
-            return new TransactionConflictException(msg);
+            newException = new TransactionConflictException(msg);
         } else if (cause instanceof TopicDoesNotExistException) {
-            return new TopicDoesNotExistException(msg);
+            newException = new TopicDoesNotExistException(msg);
         } else if (cause instanceof ProducerFencedException) {
-            return new ProducerFencedException(msg);
+            newException = new ProducerFencedException(msg);
         } else if (cause instanceof MemoryBufferIsFullError) {
-            return new MemoryBufferIsFullError(msg);
+            newException = new MemoryBufferIsFullError(msg);
         } else if (cause instanceof NotFoundException) {
-            return new NotFoundException(msg);
+            newException = new NotFoundException(msg);
         } else {
-            return new PulsarClientException(t);
+            newException = new PulsarClientException(t);
         }
+
+        Collection<Throwable> previousExceptions = getPreviousExceptions(t);
+        if (t != null) {
```

Comment:
This null check is redundant or should be put in the beginning of the method because we've already called `t`'s method before:

```java
        Throwable cause = t.getCause();
```

Reply:
it should be checking whether previousExceptions is null

Reply:
Agree.